src/include/type/varlen_pool.h

```diff
@@ -96,31 +84,11 @@ class VarlenPool {
   // Get the maximum size of this pool.
   uint64_t GetMaximumPoolSize() const;
 
-  // Get the empty buffer count for a given empty buffer list id
-  // Return -1 if list_id is out of bound
-  int GetEmptyCountByListId(size_t list_id) const;
-
-  static size_t GetRefCountSize() {
-    return sizeof(std::atomic<int64_t>);
-  }
-
-private:
-  inline std::atomic<int64_t> *GetRefCntPtr(void *addr) {
-    char *ref_cnt_addr = ((char *) addr) - sizeof(std::atomic<int64_t>);
-    return reinterpret_cast<std::atomic<int64_t> *>(ref_cnt_addr);
-  }
-
-private:
+public:
   // All these fields are implementation specific.
   // This class must be thread-safe, very very fast and provide some form of
   // compaction or garbage-collection.
 
-  // Internal memory allocation
-  void *AllocateHelper(size_t size);
-
-  // Internal memory deallocation

```

Comment:
I agree that ref count optimization can improve performance, but I think that we need to think about the design more carefully. For instance, when I switched over from `varlen pool` to global memory allocator (`nullptr`) in executor context, things broke. 
Furthermore, as you had noted in another issue, this is not a trivial change and can introduce memory leaks if not done carefully. In fact,

1) I think that none of the abstract pool implementations must clean up after themselves -- otherwise folks will start leaking memory all over the place as is already the case.

2) I am trying to fix such leaks right now in the `TileGroup`. In `SetValue`, when we overwrite an `varlen` field, we should `automatically` release the memory associated with the old value.